### PR TITLE
Remove Tensorflow Importer

### PR DESCRIPTION
**Pull request by SavyaSanchi-Sharma**
[PR OpenCV Extra](https://github.com/opencv/opencv_extra/pull/1390)
[PR OpenCV Contrib](https://github.com/opencv/opencv_contrib/pull/4171)
Removed tensorflow importers

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: samples/cpp/macbeth_chart_detection.cpp [hunk 1]
```diff
@@ -139,7 +139,7 @@ int main(int argc, char *argv[])
         if (backend != "default" || target != "cpu"){
             engine = ENGINE_CLASSIC;
         }
-        Net net = readNetFromTensorflow(model_path, pbtxt_path, engine);
+        Net net = readNet(model_path, pbtxt_path, "", engine);
```
Comment (abhishek-gola) [inline samples/cpp/macbeth_chart_detection.cpp]: You can better use ReadNetFromONNX directly

File: samples/dnn/mobilenet_ssd_accuracy.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 args = parser.parse_args()
 
 ### Get OpenCV predictions #####################################################
-net = cv.dnn.readNetFromTensorflow(cv.samples.findFile(args.weights), cv.samples.findFile(args.prototxt))
+net = cv.dnn.readNetFromONNX(cv.samples.findFile(args.weights))
 net.setPreferableBackend(cv.dnn.DNN_BACKEND_OPENCV)
```
Comment (abhishek-gola) [inline samples/dnn/mobilenet_ssd_accuracy.py]: name it args.model and add model link in description.

File: samples/python/macbeth_chart_detection.py [hunk 3]
```diff
@@ -87,12 +87,12 @@ def main(func_args=None):
...
-    if args.model and args.config:
-        # Load the DNN from TensorFlow model
+    if args.model:
+        # Load the DNN from ONNX model
         engine = cv.dnn.ENGINE_AUTO
         if args.backend != "default" or args.target != "cpu":
             engine = cv.dnn.ENGINE_CLASSIC
-        net = cv.dnn.readNetFromTensorflow(args.model, args.config, engine)
+        net = cv.dnn.readNetFromONNX(args.model, engine)
         net.setPreferableBackend(get_backend_id(args.backend))
```
Comment (asmorkalov) [inline samples/python/macbeth_chart_detection.py:93]: Please update model references in yaml file.

File: samples/python/color_correction_model.py [hunk 4]
```diff
@@ -120,12 +120,12 @@ def main(func_args=None):
...
         # Create color checker detector
-        if args.model and args.config:
-            # Load the DNN from TensorFlow model
+        if args.model:
+            # Load the DNN from ONNX model
             engine = cv.dnn.ENGINE_AUTO
             if args.backend != "default" or args.target != "cpu":
                 engine = cv.dnn.ENGINE_CLASSIC
-            net = cv.dnn.readNetFromTensorflow(args.model, args.config, engine)
+            net = cv.dnn.readNetFromONNX(args.model, engine)
```
Comment (asmorkalov) [inline samples/python/color_correction_model.py:125]: Please update model references in the yaml file.

File: modules/dnn/misc/js/gen_dict.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
     "whitelist":
     {
         "dnn_Net": ["setInput", "forward", "setPreferableBackend","getUnconnectedOutLayersNames"],
-        "": ["readNetFromTensorflow", "readNetFromTorch",
+        "": ["readNetFromTorch",
```
Comment (asmorkalov) [inline modules/dnn/misc/js/gen_dict.json]: ReadNetFromTorch should be dropped too. It's about Tourch 7, not PyTorch.

File: modules/dnn/perf/perf_net.cpp [hunk 6]
```diff
@@ -313,21 +326,29 @@ PERF_TEST_P_(DNNTestNetwork, Inception_v2_Faster_RCNN)
...
-    if ((backend == DNN_BACKEND_INFERENCE_ENGINE_NN_BUILDER_2019 && target != DNN_TARGET_CPU) ||
-        (backend == DNN_BACKEND_OPENCV && target == DNN_TARGET_OPENCL_FP16))
+    if (backend == DNN_BACKEND_INFERENCE_ENGINE_NN_BUILDER_2019 && target != DNN_TARGET_CPU)
         throw SkipTestException("");
-    processNet("dnn/faster_rcnn_inception_v2_coco_2018_01_28.pb",
-               "dnn/faster_rcnn_inception_v2_coco_2018_01_28.pbtxt",
-               cv::Size(800, 600));
+    if (backend == DNN_BACKEND_OPENCV)
+        throw SkipTestException("Faster R-CNN Inception v2: dynamic-shape ReduceMax in the TF "
+                               "Preprocessor/ResizeToRange subgraph is not yet supported by the new DNN graph engine");
```
Comment (asmorkalov) [inline modules/dnn/perf/perf_net.cpp:327]: Sounds strange. Please file a bug and put reference here, if it's true.

**Review by asmorkalov — CHANGES_REQUESTED** [review]

**Comment by abhishek-gola** [comment]
@SavyaSanchi-Sharma Rebase and fix conflicts.

**Comment by abhishek-gola** [comment]
`modules/dnn/misc/java/test/DnnTensorFlowTest.java` @SavyaSanchi-Sharma Why is this file still there?

File: modules/dnn/src/layers/conv2_layer.cpp [hunk 7]
```diff
@@ -669,6 +669,20 @@ class Conv2LayerImpl : public Conv2Layer
...
+
+            const int tail = Cout % 8;
+            if (tail) {
+                for (int n = 0; n < N; n++) {
+                    float* p = outdata + (size_t)n * out_batch_stride
+                                       + (size_t)(C1_out - 1) * (size_t)HW * 8;
+                    for (int i = 0; i < HW; i++)
+                        std::fill(p + (size_t)i * 8 + tail, p + (size_t)i * 8 + 8, 0.f);
+                }
+            }
```
Comment (abhishek-gola) [inline modules/dnn/src/layers/conv2_layer.cpp:679]: why do you need this? scatterAndActivate already applies activationFunc to the whole batch slice, so this added block applies it twice

Comment (SavyaSanchi-Sharma) [reply modules/dnn/src/layers/conv2_layer.cpp:679]: applied the fix
